tests/test_soak.py

```diff
@@ -0,0 +1,109 @@
+from utils import SPARK_PACKAGE_NAME
+
+
+LOGGER = logging.getLogger(__name__)
+TERASORT_JAR='https://downloads.mesosphere.io/spark/examples/spark-terasort-1.0-jar-with-dependencies_2.11.jar'
+TERASORT_MAX_CORES=6
+
+
+def setup_module(module):
+    _require_spark_cli()
+
+
+@pytest.mark.soak
+def test_terasort():
+    if utils.hdfs_enabled():
+        _delete_hdfs_terasort_files()
+        _run_teragen()
+        _run_terasort()
+        _run_teravalidate()
+
+
+def _require_spark_cli():
+    LOGGER.info("Ensuring Spark CLI is installed.")
+    installed_subcommands = dcos.package.installed_subcommands()
+    if any(sub.name == SPARK_PACKAGE_NAME for sub in installed_subcommands):
+        LOGGER.info("Spark CLI already installed.")
+    else:
+        LOGGER.info("Installing Spark CLI.")
+        shakedown.run_dcos_command('package install --cli {}'.format(
+            SPARK_PACKAGE_NAME))
+
+
+def _run_teragen():
+    jar_url = TERASORT_JAR
+    input_size = os.getenv('TERASORT_INPUT_SIZE', '1g')
+    utils.run_tests(jar_url,
+               "{} hdfs:///terasort_in".format(input_size),
+               "Number of records written",
+               ["--class", "com.github.ehiggs.spark.terasort.TeraGen",
+                "--conf", "spark.cores.max={}".format(TERASORT_MAX_CORES)])
+
+
+def _run_terasort():
+    jar_url = TERASORT_JAR
+    utils.run_tests(jar_url,
+               "hdfs:///terasort_in hdfs:///terasort_out",
+               "",
+               ["--class", "com.github.ehiggs.spark.terasort.TeraSort",
+                "--conf", "spark.cores.max={}".format(TERASORT_MAX_CORES)])
+
+
+def _run_teravalidate():
+    jar_url = TERASORT_JAR
+    utils.run_tests(jar_url,
+               "hdfs:///terasort_out hdfs:///terasort_validate",
+               "partitions are properly sorted",
+               ["--class", "com.github.ehiggs.spark.terasort.TeraValidate",
+                "--conf", "spark.cores.max={}".format(TERASORT_MAX_CORES)])
+
+
+def _delete_hdfs_terasort_files():
+    job_name = 'hdfs-delete-terasort-files'
+    LOGGER.info("Deleting hdfs terasort files by running job {}".format(job_name))
+    metronome_client = dcos.metronome.create_client()
+    _add_job(metronome_client, job_name)
+    _run_job_and_wait(metronome_client, job_name, timeout_seconds=300)
+    metronome_client.remove_job(job_name)
+    LOGGER.info("Job {} completed".format(job_name))
+
+
+def _add_job(metronome_client, job_name):
+    jobs_folder = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), 'jobs', 'json'
+    )
+    job_path = os.path.join(jobs_folder, '{}.json'.format(job_name))
+    with open(job_path) as job_file:
+        job = json.load(job_file)
+    metronome_client.add_job(job)
+
+
+def _run_job_and_wait(metronome_client, job_name, timeout_seconds):
+    metronome_client.run_job(job_name)
+
+    shakedown.wait_for(
+        lambda: (
+            'Successful runs: 1' in
+            _run_cli('job history {}'.format(job_name))
```

Comment:
Checking a job's completion status is messy in the CLI; it hits a `jobs` endpoint and parses the results: https://github.com/dcos/dcos-cli/blob/0.4.16/cli/dcoscli/job/main.py#L266 I will use the CLI for now.